modules/message.sh

```diff
@@ -275,20 +286,52 @@ function message {
-            echo "Then install APP to the workspace and click the allow button to give permissions in the redirected link and then you will get the \"OAuth access token\""
-            echo "Run $BASENAME message slack apitoken <oauth access token>"
+            echo "1. Go to https://api.slack.com/apps"
+            echo "2. Click \"Create New App\""
+            echo "3. Type a name in \"App Name\""
+            echo "4. Select the \"Development Slack Workspace\" (eg : Open Learning Exchange)\ from the drop-down"
+            echo "5. Click \"OAuth & Permission\" under \"features \""
+            echo "6. Under \"User Token Scopes\", click \"Add an OAuth Scope\""
+            echo "7. Add the following permissions:"
+            echo "  \"chat:write\""
+            echo "  \"channels:write\""
+            echo "  \"channels:read\""
+            echo "  \"channels:history\""
+            echo "  \"groups:write\""
+            echo "  \"groups:read\""
+            echo "  \"im:write\""
+            echo "  \"mpim:write\""
+            echo "  \"usergroups:read\""
+            echo "  \"users:read\""
+            echo "8. Under \"OAuth Tokens & Redirect URLs\", click \"Install to Workspace\""
+            echo "9. Click \"Allow\". This will give permissions and display the User OAuth Token"
+            echo "10. Run $BASENAME message slack apitoken <User OAuth Token>"
           fi
           ;;
         channels)
           if check_apitoken slack; then
-            channel_names=$(get_channel_slack)
-            echo "Channels Names:"
-            echo
-            echo "$channel_names"
+            if channel_names=$(get_channel_slack); then
+              echo "Channels Names:"
+              echo
+              echo "$channel_names"
+            else
+              echo "Error: Failed to use the following permissions:"
+              if curl -s -F token=$access_token -F types=public_channel,private_channel https://slack.com/api/users.conversations | grep -q "channels:read"; then
+                echo "  channels:read"
+              fi
+              if curl -s -F token=$access_token -F types=public_channel,private_channel https://slack.com/api/users.conversations | grep -q "groups:read"; then
+                echo "  groups:read"
+              fi
+              if curl -s -F token=$access_token -F types=public_channel,private_channel https://slack.com/api/users.conversations | grep -q "mpim:read"; then
+                echo "  mpim:read"
+              fi
+              if curl -s -F token=$access_token -F types=public_channel,private_channel https://slack.com/api/users.conversations | grep -q "im:read"; then
```

Comment:
Wouldn't it be better if you replace `grep -q "<insert scope>` with `jq .<insert scope>`.

Reply:
Ready for another review.